webpack/redux/reducers/RedHatRepositories/enabled.js

```diff
@@ -57,40 +58,46 @@ const changeRepoState = (state, repoToChange, stateDiff) => (
   )
 );
 
+// eslint-disable-next-line consistent-return
 export default (state = initialState, action) => {
-  if (action.type === DISABLE_REPOSITORY_REQUEST) {
-    return changeRepoState(state, action.repository, { loading: true });
-  } else if (action.type === DISABLE_REPOSITORY_FAILURE) {
-    return changeRepoState(state, action.payload.repository, { loading: false });
-  } else if (action.type === ENABLED_REPOSITORIES_REQUEST) {
-    if (!action.silent) {
-      return state.set(['loading'], true);
-    }
-  } else if (action.type === ENABLED_REPOSITORIES_SUCCESS) {
-    const {
-      page, per_page, subtotal, results, // eslint-disable-line camelcase
-    } = action.response;
-
-    return Immutable({
-      repositories: mapRepositories(results),
-      pagination: {
-        page: Number(page) || 1,
-        // server can return per_page: null when there's error in the search query,
-        // don't store it in such case
-        // eslint-disable-next-line camelcase
-        perPage: Number(per_page || state.pagination.perPage),
-      },
-      itemCount: Number(subtotal),
-      loading: false,
-      searchIsActive: !isEmpty(action.search),
-      search: action.search,
-    });
-  } else if (action.type === ENABLED_REPOSITORIES_FAILURE) {
-    return Immutable({
-      repositories: [],
-      loading: false,
-      error: action.error,
-    });
+  let page;
+  let per_page; // eslint-disable-line camelcase
+  let subtotal;
+  let results;
+  switch (action.type) {
+    case DISABLE_REPOSITORY_REQUEST:
+      return changeRepoState(state, action.repository, { loading: true });
+    case DISABLE_REPOSITORY_FAILURE:
+      return changeRepoState(state, action.payload.repository, { loading: false });
+    case ENABLED_REPOSITORIES_REQUEST:
+      if (!action.silent) {
+        return state.set(['loading'], true);
+      }
+      return state;
+    case ENABLED_REPOSITORIES_SUCCESS:
+      ({ page, per_page, subtotal, results } = action.response); // eslint-disable-line
```

Comment:
How does changing `const` to `let` help in this and similar case? Asking cause we do use `const` in some other reducers..

Reply:
@sjha4 This was just me fighting with ESLint. 
1.  First, I tried using `const` in the case statement.  It I wasn't allowed to have a declaration within the case statement.
2. Then, I moved the `const` declaration up to the top.  But since those values aren't always defined, it threw errors.
3. So, I ended up declaring them at the top and assigning the values in the case.

Reply:
Can you use `const` with that line disabled?  Also if we're going to disable this line shouldn't we disable just for that particular rule?  @sharvit @amirfefer any other ideas here?

Reply:
@waldenraines  Not sure what I was doing before, but today I was able to get Eslint to use `const` in the case statement instead, which is much better.  thanks for the suggestion!